contracts/deploy/mainnet/126_pool_booster_curve.js

```diff
@@ -0,0 +1,236 @@
+    let implementationAddress = ethers.utils.getAddress(
+      `0x${txReceipt.events[1].topics[1].slice(26)}`
+    );
+    console.log(
+      `Curve Booster Implementation OETH deployed at: ${implementationAddress}`
+    );
+
+    // --- Deploy and init proxy --- //
+    let cachedInitCodeProxy = ProxyBytecode.bytecode; // No constructor arguments
+    let initializeImplem = cCurvePoolBooster.interface.encodeFunctionData(
+      "initialize(address,uint16,address,address,address)",
+      [
+        addresses.multichainStrategist, // strategist
+        fee, // fee
+        addresses.multichainStrategist, // feeCollector
+        addresses.mainnet.CampaignRemoteManager, // campaignRemoteManager
+        addresses.votemarket, // votemarket
+      ]
+    );
+    let initializeProxy = cCurvePoolBoosterProxy.interface.encodeFunctionData(
+      "initialize(address,address,bytes)",
+      [
+        implementationAddress, // implementation
+        addresses.mainnet.Timelock, // governor
+        initializeImplem, // init data
+      ]
+    );
+    let txResponseProxy = await withConfirmation(
+      cCreateX
+        .connect(sDeployer)
+        .deployCreate2AndInit(
+          encodedSalt,
+          cachedInitCodeProxy,
+          initializeProxy,
+          ["0x00", "0x00"],
+          deployerAddr
+        )
+    );
+    let txReceiptProxy = await txResponseProxy.wait();
+    // event 0 is GovernorshipTransferred
+    // event 1 is ContractCreation, topics[1] is the address of the deployed contract, topics[2] is the salt
+    // event 2 is StrategistUpdated
+    // event 3 is FeeUpdated
+    // event 4 is FeeCollectorUpdated
+    // event 5 is CampaignRemoteManagerUpdated
+    // event 6 is GovernorshipTransferred
+    const pbOETHProxyAddress = ethers.utils.getAddress(
+      `0x${txReceiptProxy.events[1].topics[1].slice(26)}`
+    );
+    console.log(`Curve Booster Proxy OETH deployed at: ${pbOETHProxyAddress}`);
+
+    // ------------------------------------------------------------------------------------------------------------------------------------
+    // ---
+    // --- OUSD Pool Booster
+    // ---
+    // ------------------------------------------------------------------------------------------------------------------------------------
+    console.log("\nStarting deployment using CreateX: OUSD");
+    rewardToken = addresses.mainnet.OUSDProxy;
+    gauge = addresses.mainnet.CurveTriOGNGauge;
+    targetedChainId = 42161; // arbitrum
+
+    // Generate salt
+    salt = ethers.utils.keccak256(
+      ethers.utils.concat([
+        ethers.utils.arrayify(rewardToken),
+        ethers.utils.arrayify(gauge),
+        ethers.utils.arrayify(1),
+      ])
+    );
+
+    encodedSalt = encodeSaltForCreateX(deployerAddr, false, salt);
+    console.log(`Encoded salt for OUSD: ${encodedSalt}`);
+
+    // --- Deploy implementation --- //
+    cachedInitCodeImpl = ethers.utils.concat([
```

Comment:
we don't need to deploy the implementation contracts on the same address across the chain right? only the proxy address should match

Reply:
Yes, this is not mandatory. But I think it looks prettier :)

Reply:
Agreed might look prettier since the implementation address is the same, but it makes the deploy files more complex to write and to review.